### PR TITLE
Do not alter paths data when loading through d3js

### LINKED ISSUE #7021: Cannot Save Paths
## Bug description

Cannot save paths because there is a circular reference in the JSON to be saved

## Expected behavior

Paths save

## How to reproduce

1. go to a new or existing Paths insight locally or on cloud
2. try to save it
3. see that nothing happens
4. go to the console and see the error

## Environment

- [X] PostHog Cloud
- [ ] self-hosted PostHog (ClickHouse-based), version/commit: _please provide_
- [ ] self-hosted PostHog (Postgres-based, legacy), version/commit: _please provide_

### PR DESCRIPTION
**Pull request by pauldambra**
## Changes

closes #7021 and #6934 

## How did you test this code?

by running it locally and showing I can save the data with this fix in place

### CODE REVIEW

File: frontend/src/scenes/paths/NewPaths.tsx [hunk 1]
```diff
@@ -271,7 +271,10 @@ export function NewPaths({ dashboardItemId = null, color = 'white' }: PathsProps
 
         const svg = createCanvas(width, height)
         const sankey = createSankey(width, height)
-        const { nodes, links } = sankey(paths)
+        const { nodes, links } = sankey({
+            nodes: paths.nodes.map((d) => ({ ...d })),
```
Comment (hazzadous) [inline frontend/src/scenes/paths/NewPaths.tsx:275]: non-blocking

Should we add in `...paths` as well to ensure that any other attributes that were considered and part of paths are also considered by `sankey`.  I have no idea if that's a likely scenario

**Review by hazzadous — APPROVED** [review]
Looks good

File: frontend/src/scenes/paths/NewPaths.tsx (hunk 1 above)
Comment (pauldambra) [reply frontend/src/scenes/paths/NewPaths.tsx:275]: Reading the code in d3-sankey I'm pretty sure it's only node and links that are used by sankey

(We also explode paths the same way in the Old `Paths.tsx`)

Comment (pauldambra) [reply frontend/src/scenes/paths/NewPaths.tsx:275]: sankey code here https://github.com/d3/d3-sankey/blob/master/src/sankey.js

Comment (pauldambra) [reply frontend/src/scenes/paths/NewPaths.tsx:275]: and our older usage here https://github.com/PostHog/posthog/blob/master/frontend/src/scenes/paths/Paths.jsx#L139

**Pull request merged**